src/classes/vocabulary/jsonld/class-term-jsonld.php

```diff
@@ -39,4 +69,94 @@ public function wl_term_jsonld_array( $jsonld_array, $term_id ) {
 		return $jsonld_array;
 	}
 
+	/**
+	 * Set term jsonld array event.
+	 *
+	 * @param $jsonld_arr array The final jsonld before outputting to page.
+	 * @param $term_id int The term id for which the jsonld is generated.
+	 * @param $context int A context for the JSON-LD generation, valid values in Jsonld_Context_Enum
+	 *
+	 * @return array
+	 */
+	public function wl_term_jsonld_array_event( $jsonld_arr, $term_id, $context ) {
+		// If context is not PAGE or the array is empty, return early.
+		if ( Jsonld_Context_Enum::PAGE !== $context || empty( $jsonld_arr[0] ) ) {
+			return $jsonld_arr;
+		}
+
+		// Flag to indicate if we should make an API request.
+		$change_status = false;
+
+		// Get data from the array.
+		$data = $jsonld_arr[0];
+
+		// Fetch the initial 'about' and 'mentions' counts from term meta.
+		$counts = array(
+			'about'    => get_term_meta( $term_id, 'wl_about_count', true ) ? get_term_meta( $term_id, 'wl_about_count', true ) : 0,
+			'mentions' => get_term_meta( $term_id, 'wl_mentions_count', true ) ? get_term_meta( $term_id, 'wl_mentions_count', true ) : 0,
+		);
+
+		// Iterate over the counts array.
+		foreach ( $counts as $key => $count ) {
+			// Check if data has 'about' or 'mentions' and the count is different from the existing meta value.
+			if ( ! empty( $data[ $key ] ) ) {
+				$new_count = count( $data[ $key ] );
+				if ( $count !== $new_count ) {
+					// Set flag to true if counts have changed.
+					$change_status = true;
+
+					// Update the counts array with new count.
+					$counts[ $key ] = $new_count;
+
+					// Update term meta with new count.
+					update_term_meta( $term_id, 'wl_' . $key . '_count', $new_count );
+				}
+			}
+		}
+
+		// If the count has changed, make the API request.
+		if ( $change_status ) {
+			$this->api_service->request(
+				'POST',
+				'/plugin/events',
+				array( 'Content-Type' => 'application/json' ),
+				wp_json_encode(
+					array(
+						'source' => 'jsonld',
+						'args'   => array(
+							array( 'about_count' => $counts['about'] ),
+							array( 'mentions_count' => $counts['mentions'] ),
+						),
+						'url'    => $this->get_term_url( $term_id ),
+					)
+				),
+				0.001,
+				null,
+				array( 'blocking' => false )
+			);
+		}
+
+		return $jsonld_arr;
+	}
+
+	/**
+	 * Get term url.
+	 *
+	 * @param $id
+	 *
+	 * @return array|false|int|mixed|string|\WP_Error|\WP_Term|null
+	 */
+	private function get_term_url( $id ) {
+		if ( null === $id ) {
+			return isset( $_SERVER['REQUEST_URI'] ) ? filter_var( wp_unslash( $_SERVER['REQUEST_URI'] ), FILTER_SANITIZE_URL ) : '';
+		}
+
+		$maybe_url = get_term_meta( $id, Wordlift_Url_Property_Service::META_KEY, true );
+		if ( ! empty( $maybe_url ) ) {
+			return $maybe_url;
+		}
+
+		return get_term_link( $id );
```

Comment:
We should really get the current page here, I am not sure what's happening here?